dashboard/src/components/ModuleCard.jsx

```diff
@@ -0,0 +1,246 @@
+import { requestData } from "../utils/rest";
+import { ModulesContext } from "./DashboardContext";
+import ModuleSettings from "./ModuleSettings";
+
+import ExternalLink from "./ExternalLink";
+import { renderToString, useContext, useState } from "@wordpress/element";
+import { __ } from "@wordpress/i18n";
+import { ExternalLinkIcon, MoveUpRight } from "lucide-react";
+import { toaster } from "./ui/toaster";
+import ObfxBadge from "./ObfxBadge";
+
+const { root, toggleStateRoute, options, menusSupport } = obfxDash;
+
+const ModuleCard = ({ slug, details }) => {
+  const refreshAfterEnabled = details.refresh_after_enabled;
+  const activeDefault = details.active_default;
+  const documentationUrl = details.documentation_url;
+  const [loading, setLoading] = useState(false);
+  const { modulesData, setModulesData } = useContext(ModulesContext);
+  const moduleStatus = modulesData.module_status;
+
+  const updateModuleStatus = ({ checked }) => {
+    setLoading(true);
+
+    const dataToSend = { slug, value: checked };
+    requestData(root + toggleStateRoute, false, dataToSend).then((r) => {
+      toggleStatusCallback(r, checked);
+    });
+  };
+
+  const toggleStatusCallback = (r, value) => {
+    if (r.type !== "success") {
+      setLoading(false);
+      toaster.error({
+        title: __(
+          "Could not activate module. Please try again.",
+          "themeisle-companion"
+        ),
+      });
+      return;
+    }
+
+    if (refreshAfterEnabled) {
+      window.location.reload();
+    }
+
+    if (!moduleStatus[slug]) {
+      moduleStatus[slug] = {};
+    }
+
+    moduleStatus[slug].active = value;
+    setModulesData(modulesData);
+    setLoading(false);
+    toaster.success({
+      title:
+        (value
+          ? __("Module activated", "themeisle-companion")
+          : __("Module deactivated", "themeisle-companion")) +
+        ` (${details.name})`,
+    });
+  };
+
+  const renderDescription = (description) => {
+    const elements = [];
+    if (description.indexOf("neve-pro-notice") >= 0) {
+      return (
+        <Text
+          fontSize="sm"
+          lineHeight="1.6"
+          m="0"
+          color="fg.muted"
+          dangerouslySetInnerHTML={{ __html: description }}
+        />
+      );
+    }
+
+    while (description.indexOf("<a") >= 0) {
+      const start = description.indexOf("<a");
+      const end = description.indexOf("</a>");
+
+      elements.push(description.slice(0, start));
+
+      const hrefStart = description.indexOf('href="') + 'href="'.length;
+      const hrefEnd = description.indexOf('"', hrefStart);
+      const href = description.slice(hrefStart, hrefEnd);
+
+      const anchorText = description.slice(
+        description.indexOf(">", start) + 1,
+        end
+      );
+
+      elements.push(
+        renderToString(<ExternalLink href={href}>{anchorText}</ExternalLink>)
+      );
+
+      description = description.slice(end + "</a>".length);
+    }
+
+    elements.push(description);
```

Comment:
If the description is simple, I think we can simplify this to:

```js
const parseDescription = (description) => {
  const parser = new DOMParser();
  const doc = parser.parseFromString(description, 'text/html');
  const elements = [];
  
  doc.body.childNodes.forEach(node => {
    if (node.nodeType === Node.TEXT_NODE) {
      elements.push(node.textContent);
    } else if (node.nodeName === 'A') {
      elements.push(
        renderToString(
          <ExternalLink href={node.href}>{node.textContent}</ExternalLink>
        )
      );
    }
  });
  
  return elements;
};
```

Reply:
I am not sure how complex descriptions get, but will keep this in mind while we iterate.